_ml-commons-plugin/connectors.md

```diff
@@ -0,0 +1,461 @@
+                  "content": "Hello! How can I assist you today?"
+                },
+                "finish_reason": "stop"
+              }
+            ],
+            "usage": {
+              "prompt_tokens": 19,
+              "completion_tokens": 9,
+              "total_tokens": 28
+            }
+          }
+        }
+      ]
+    }
+  ]
+}
+```
+
+### Internal connector
+
+To create an internal connector, add the `connector` parameter to the Register model API, as shown in the following example:
+
+```json
+POST /_plugins/_ml/models/_register
+{
+    "name": "openAI-GPT-3.5 completions: internal connector",
+    "function_name": "remote",
+    "model_group_id": "lEFGL4kB4ubqQRzegPo2",
+    "description": "test model",
+    "connector": {
+        "name": "OpenAI Connector",
+        "description": "The connector to public OpenAI model service for GPT 3.5",
+        "version": 1,
+        "protocol": "http",
+        "parameters": {
+            "endpoint": "api.openai.com",
+            "max_tokens": 7,
+            "temperature": 0,
+            "model": "text-davinci-003"
+        },
+        "credential": {
+            "openAI_key": "..."
+        },
+        "actions": [
+            {
+                "action_type": "predict",
+                "method": "POST",
+                "url": "https://${parameters.endpoint}/v1/completions",
+                "headers": {
+                    "Authorization": "Bearer ${credential.openAI_key}"
+                },
+                "request_body": "{ \"model\": \"${parameters.model}\", \"prompt\": \"${parameters.prompt}\", \"max_tokens\": ${parameters.max_tokens}, \"temperature\": ${parameters.temperature} }"
+            }
+        ]
+    }
+}
+```
+
+
+## Examples 
+
+The following example connector requests show how to create a connector with supported third-party tools.
+
+
+### OpenAI chat connector
+
+The following example creates a standalone OpenAI chat connector. The same options can be used for an internal connector under the `connector` parameter:
+
+
+```json
+POST /_plugins/_ml/connectors/_create
+{
+    "name": "OpenAI Chat Connector",
+    "description": "The connector to public OpenAI model service for GPT 3.5",
+    "version": 1,
+    "protocol": "http",
+    "parameters": {
+        "endpoint": "api.openai.com",
+        "model": "gpt-3.5-turbo"
+    },
+    "credential": {
+        "openAI_key": "..."
+    },
+    "actions": [
+        {
+            "action_type": "predict",
+            "method": "POST",
+            "url": "https://${parameters.endpoint}/v1/chat/completions",
+            "headers": {
+                "Authorization": "Bearer ${credential.openAI_key}"
+            },
+            "request_body": "{ \"model\": \"${parameters.model}\", \"messages\": ${parameters.messages} }"
+        }
+    ]
+}
+```
+
+After creating the connector, you can retrieve the `task_id`, deploy the model, and use the Predict API, similar to a standalone connector.
```

Comment:
> After creating the connector, you can retrieve the `task_id`, deploy the model, and use the Predict API, similar to a standalone connector.

-> 

After creating the connector, you can use the connector id to register model, deploy and predict.